src/WebApp/Services/BasketState.cs

```diff
@@ -14,22 +17,45 @@ public class BasketState(
 {
     private Task<IReadOnlyCollection<BasketItem>>? _cachedBasket;
     private HashSet<BasketStateChangedSubscription> _changeSubscriptions = new();
-
+    private HttpContext? HttpContext { get; set; }
+    Dictionary<int,int>? ProductIdToQuantity = new();
     public Task DeleteBasketAsync()
         => basketService.DeleteBasketAsync();
 
     public async Task<IReadOnlyCollection<BasketItem>> GetBasketItemsAsync()
         => (await GetUserAsync()).Identity?.IsAuthenticated == true
         ? await FetchBasketItemsAsync()
         : [];
-
+    public async Task<IReadOnlyCollection<BasketItem>> GetBasketItemsAsAnonymous(HttpContext HttpContext)
+        => (await GetUserAsync()).Identity?.IsAuthenticated == false
+        ? await GetBasketItemsAsAnonymousAsync(HttpContext)
+        : [];
     public IDisposable NotifyOnChange(EventCallback callback)
     {
         var subscription = new BasketStateChangedSubscription(this, callback);
         _changeSubscriptions.Add(subscription);
         return subscription;
     }
 
+    public async Task AddAsAnonymousUser(CatalogItem item,HttpContext HttpContext)
```

Comment:
Instead of passing the entire `HttpContext`, can these APIs just consume the session value?

Reply:
Dependency injected using IHttpContextAccessor and used session variable in this file. Please review again.